core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/scheduling/resourcePolicies/HistoricalRuntimeStats.scala

```diff
@@ -0,0 +1,62 @@
+package edu.uci.ics.amber.engine.architecture.scheduling.resourcePolicies
+
+import edu.uci.ics.amber.engine.common.virtualidentity.PhysicalOpIdentity
+import edu.uci.ics.texera.web.SqlServer
+import edu.uci.ics.texera.web.model.jooq.generated.tables.daos.{
+  WorkflowDao,
+  WorkflowExecutionsDao,
+  WorkflowRuntimeStatisticsDao
+}
+import edu.uci.ics.texera.web.model.jooq.generated.tables.pojos.WorkflowRuntimeStatistics
+import org.jooq.DSLContext
+
+import java.io.{BufferedWriter, FileWriter}
+import scala.collection.mutable
+import scala.jdk.CollectionConverters._
+
+class HistoricalRuntimeStats() {
+  lazy val context: DSLContext = SqlServer.createDSLContext()
+  val workflowRuntimeStatisticsDao = new WorkflowRuntimeStatisticsDao(context.configuration)
+  val workflowDao = new WorkflowDao(context.configuration)
+  val workflowExecutionsDao = new WorkflowExecutionsDao(context.configuration())
+
+  def getHistoricalRuntimeStatsByOperatorVersion(
+      physicalOpIdentity: PhysicalOpIdentity
+  ): mutable.Buffer[WorkflowRuntimeStatistics] = {
+    val historicalRuntimeStats =
+      workflowRuntimeStatisticsDao.fetchByOperatorId(physicalOpIdentity.logicalOpId.id)
+
+    val file = new BufferedWriter(new FileWriter("history.txt"))
+
+    try {
+      file.write(historicalRuntimeStats.size.toString)
+      historicalRuntimeStats.forEach { key =>
+        file.write(key.getOperatorId)
+        file.write(" ")
+        file.write(key.getExecutionId.toString)
+        file.write(" ")
+        file.write(key.getWorkflowId.toString)
+        file.write(" ")
+        file.newLine()
+      }
+    } finally {
+      file.close()
+    }
```

Comment:
do not use files as we need to manage its lifecycle. you need to find a better way to pass information. Also, you need a better format for this information. I suggest you use JSON format.